Sources/Authentication/Passwords/Bcrypt.swift

```diff
@@ -0,0 +1,257 @@
+internal import CVaporAuthBcrypt
+
+#if canImport(FoundationEssentials)
+public import Foundation
+#else
+public import Foundation
+#endif
+
+// MARK: bcrypt
+
+/// Creates and verifies bcrypt hashes.
+///
+/// Use BcryptDigest to create hashes for sensitive information like passwords.
+///
+///     try BcryptDigest.hash("vapor", cost: 12)
+///
+/// bcrypt uses a random salt each time it creates a hash. To verify hashes, use the `verify(_:matches)` method.
+///
+///     let hash = try BcryptDigest.hash("vapor", cost: 12)
+///     try BcryptDigest.verify("vapor", created: hash) // true
+///
+/// https://en.wikipedia.org/wiki/bcrypt
+public enum BcryptDigest: Sendable {
```

Comment:
Not a huge fan of `BcryptDigest`, I'd be more for `Bcrypt` has this type is really just a namespace and doesn't represent a digest itself. (I'm also biased because we could switch it out for https://github.com/ptoffy/swift-bcrypt once it reaches C performance)

Reply:
Will rename it to `VaporBcrypt` - I'm trying to avoid polluting the global namespace somewhat, and we can always rename it before release